### PR TITLE
Fix the FifoQueue option for the Queue shortcut

### LINKED ISSUE #94: FifoQueue option is not usable in Queue shortcut
If you try to use the queue shortcut's `FifoQueue` option, you are likely to run into an error like this one:

> CREATE_FAILED Queue: The name of a FIFO queue can only include alphanumeric characters, hyphens, or underscores, must end with .fifo suffix and be 1 to 80 in length.

I think it would be reasonable for `QueueName` to add the `.fifo` suffix to its default value (`${AWS::StackName}-${LogicalName}`) when `FifoQueue: true`.

### PR DESCRIPTION
**Pull request by davidtheclark**
Closes #94.

I still need to test this by deploying a real stack with it, so I'll leave the PR in draft mode until then.

### CODE REVIEW

**Comment by davidtheclark** [comment]
I've successfully deployed a few FIFO queues with now, so this is ready for review, @mapbox/release-engineering.

File: lib/shortcuts/queue.js [hunk 1]
```diff
@@ -102,60 +111,65 @@ class Queue {
         Properties: {
           MessageRetentionPeriod: 1209600,
           VisibilityTimeout: DeadLetterVisibilityTimeout,
-          QueueName: {
-            'Fn::Sub': ['${queue}-dead-letter', { queue: QueueName }]
-          }
+          // The dead-letter queue's type must match the main queue's type.
+          FifoQueue,
```
Comment (rclark) [inline lib/shortcuts/queue.js:115]: `if (FifoQueue === false)`, then the deploy will fail. `undefined` is fine here, but we should add a check somewhere that converts a user-provided `false` into `undefined`.

Comment (davidtheclark) [reply lib/shortcuts/queue.js:117]: Ah thanks for the reminder. Added that in f1696ea.

**Review by rclark — APPROVED** [review]

**Comment by davidtheclark** [comment]
Published in 4.2.2.

**Pull request merged**